src/app/core/pipe/filter.pipe.ts

```diff
@@ -0,0 +1,19 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Course} from '../models/course';
+
+@Pipe({
+  name: 'filter'
+})
+
+export class FilterPipe implements PipeTransform {
+  transform(data: Array<Course>, term: string): Array<Course> {
+    if (!data) {
+      return [];
+    }
+    if (!term) {
+      return data;
+    }
+
+    return data.filter((item: Course) => JSON.stringify(item).toLowerCase().includes(term.toLowerCase()));
```

Comment:
would be great to add 'trim' method for this one

Reply:
done, thanks for your comments.